Driver does not compile on kernel 5.6

```
In file included from /scratch/source/gpcieuni_drv.c:4:
/scratch/source/pcieuni_io.h:28:26: error: field 'start_time' has incomplete type
   28 |         struct timeval   start_time;
      |                          ^~~~~~~~~~
/scratch/source/pcieuni_io.h:29:26: error: field 'stop_time' has incomplete type
   29 |         struct timeval   stop_time;
      |                          ^~~~~~~~~
make[6]: *** [/usr/src/linux-5.6.0-1/scripts/Makefile.build:269: /scratch/source/gpcieuni_drv.o] Error 1
make[5]: *** [/usr/src/linux-5.6.0-1/Makefile:1698: /scratch/source] Error 2
make[4]: *** [../../../linux-5.6.0-1/Makefile:180: sub-make] Error 2
make[4]: Leaving directory '/usr/src/linux-5.6.0-1-obj/x86_64/default
```

--- a/pcieuni_io.h
+++ b/pcieuni_io.h
@@ -3,6 +3,7 @@
 
 #include <linux/types.h>
 #include <linux/ioctl.h> /* needed for the _IOW etc stuff used later */
+#include <linux/version.h>
 
 #define PCIEUNI_DMA_SYZE                    4096
 
@@ -24,9 +25,24 @@ struct device_ioctrl_dma  {
 };
 typedef struct device_ioctrl_dma device_ioctrl_dma;
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,6,0)
+
+typedef struct timeval pcieuni_timeval;
+
+#else
+
+struct pcieuni_timeval {
+	long long tv_sec;
+	long long tv_usec;
+};
+
+typedef struct pcieuni_timeval pcieuni_timeval;
+#endif
+
 struct device_ioctrl_time  {
-        struct timeval   start_time;
-        struct timeval   stop_time;
+
+        pcieuni_timeval   start_time;
+        pcieuni_timeval   stop_time;
 };
 typedef struct device_ioctrl_time device_ioctrl_time;
 

--- a/pcieuni_ufn.h
+++ b/pcieuni_ufn.h
@@ -260,10 +260,16 @@ void unregister_gpcieuni_proc(int num, char *dfn);
     int        pcieuni_procinfo(char *, char **, off_t, int, int *,void *);
 #else
     ssize_t pcieuni_procinfo(struct file *filp,char *buf,size_t count,loff_t *offp );
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,6,0)
     static const struct file_operations gpcieuni_proc_fops = { 
         .read = pcieuni_procinfo,
     }; 
-#endif
+#else
+    static const struct proc_ops gpcieuni_proc_fops = {
+        .proc_read = pcieuni_procinfo,
+    };
+#endif   /* KERNEL_VERSION(5,6,0) */
+#endif   /* KERNEL_VERSION(3,10,0) */
 
 #endif	/* PCIEUNI_UFN_H */
 

